lib/components/viewers/RouteRow.js

```diff
@@ -138,26 +200,29 @@ export class RouteRow extends PureComponent {
     }
   }
 
-  // Used to ensure details are visible until animation completes
-  setNewHeight = (newHeight) => {
-    this.setState({ newHeight })
+  _patternButtonClicked = () => {
+    const { route, setViewedRoute } = this.props
+    const { id, patterns } = route
+    const firstPattern = route && patterns && Object.values(patterns)?.[0]?.id
+    setViewedRoute({ patternId: firstPattern, routeId: id })
```

Comment:
Does `setViewedRoute` require a pattern id? It has all the pattern geometry, shouldn't it be able to render all of the patterns at once?

Reply:
Yep! Passing the pattern id is what takes it to the pattern viewer. The `_onClick` for the `RouteRowButton` renders all the patterns.

Reply:
Oh sorry! Slept right through that one!